main/network.c

```diff
@@ -1292,16 +1295,28 @@ struct hostent * gethostname_re (const char *host,struct hostent *hostbuf,char *
 	while (( res = 
 		gethostbyname_r(host,hostbuf,*tmphstbuf,*hstbuflen,&hp,&herr))
 		&& (errno == ERANGE)) {
+		char *ptmphstbuf;
 		/* Enlarge the buffer. */
+		if (*hstbuflen * 2 >= SIZE_MAX) {
```

Comment:
This condition is always false. Should probably be `*hstbuflen > SIZE_MAX / 2`? (Same below)